lib/mail/elements/received_element.rb

```diff
@@ -27,7 +27,7 @@ def to_s(*args)
       def datetime_for(received)
         ::DateTime.parse("#{received.date} #{received.time}")
       rescue ArgumentError => e
-        raise e unless e.message == 'invalid date'
+        raise unless e.message =~ /\A(invalid date|string length \(\d+\) exceeds the limit \d+)\z/
```

Comment:
Let's call the utility method you will create.  One place to keep this up to date makes sense.